app/src/main/java/org/mozilla/fenix/customtabs/ExternalAppBrowserActivity.kt

```diff
@@ -57,6 +65,12 @@ open class ExternalAppBrowserActivity : HomeActivity() {
         }
     }
 
+    override fun onPause() {
+        super.onPause()
+
+        engineView.release()
```

Comment:
Does this need to be in `onPause` instead of `onStop` or `onDestroy`? I'm concerned about cases like opening the notification tray, or focusing on another app in multi-window mode. If this activity is resumed, do we recapture the engine view?

Reply:
In my tests `onStop` was not called before `onStart` was called in the new activity, this is the root of the issue, because the session would be released `onStop`. But engine view will automatically be recovered `onStart` and `onResume`. Notification drawer works without issues. Split screen also mostly works, only if I go to the homecreen while in split screen and then return to the split screen view, the WebApp is blank. I will look at this again.

Reply:
I tested this again and `onStop` does really fit better here. Only when I put a PWA in split screen, go to the launcher and then launch the same PWA again, Fenix crashes. It appears android is not calling `onStop` in that particular situation.

Reply:
@NotWoods I think this is best solved by releasing the session through it's observer during the activity creation. I implemented this here https://github.com/mozilla-mobile/android-components/commit/b6cd8418a2742a9fa0788348dc3b0731624c0f96. What do you think? Should I create a new AC issue + PR for this?

Reply:
Yeah, please file a new issue and PR :)

Reply:
@NotWoods PR is here: https://github.com/mozilla-mobile/android-components/pull/7381